src/syft/core/plan/translation/torchscript/plan_translate.py

```diff
@@ -0,0 +1,108 @@
+# stdlib
+from collections import OrderedDict
+from typing import Any
+from typing import List
+
+# third party
+import torch as th
+
+# syft absolute
+from syft.core.plan import Plan
+from syft.core.plan.plan_builder import PLAN_BUILDER_VM
+from syft.core.plan.plan_builder import ROOT_CLIENT
+from syft.lib.python.collections import OrderedDict as SyOrderedDict
+from syft.lib.python.dict import Dict
+from syft.lib.python.list import List as SyList
+from syft.logger import traceback_and_raise
+
+# syft relative
+from ....pointer.pointer import Pointer
+from .plan import PlanTorchscript
+
+
+def translate(plan: Plan) -> PlanTorchscript:
+    """Translates Syft Plan to torchscript"""
+
+    class PlanWrapper(th.nn.Module):
+        """
+        Plan needs to be executed with kwargs while torchscript needs to be executed with args,
+        so we need to have kwarg names inside torchscript to pass them into Plan.
+        We use nn.Module to store input kwarg names inside torchscript.
+        """
+
+        def __init__(self, kwarg_names: List[str]):
+            _kwarg_names: List[str]
+            super(PlanWrapper, self).__init__()
+            self._kwarg_names = kwarg_names
+
+        def forward(self, *args: Any) -> Any:
+            kwarg_ptrs: Dict[str, Pointer] = {}
+
+            # Since Syft Plan needs pointers as args,
+            # reverse-map actual arg values to pointers
+            for name, arg in zip(self._kwarg_names, args):
+                found = False
+                for ptr in ROOT_CLIENT.store:
```

Comment:
It would be great if we could somehow simplify this search. The whole part of Actions working with pointers is also somewhat annoying in the normal `Plan` class. Maybe for now extract this complexity in a function/classmethod to improve readability?